app/src/main/java/com/kylecorry/trail_sense/tools/survival_guide/ui/SurvivalGuideBottomSheetFragment.kt

```diff
@@ -0,0 +1,148 @@
+
+import android.content.res.ColorStateList
+import android.graphics.Color
+import android.view.View
+import android.widget.TextView
+import androidx.appcompat.widget.AppCompatImageButton
+import androidx.core.os.bundleOf
+import androidx.core.view.isVisible
+import androidx.core.widget.NestedScrollView
+import com.kylecorry.andromeda.core.system.Resources
+import com.kylecorry.andromeda.core.ui.Colors
+import com.kylecorry.andromeda.core.ui.setCompoundDrawables
+import com.kylecorry.andromeda.core.ui.useService
+import com.kylecorry.andromeda.markdown.MarkdownService
+import com.kylecorry.andromeda.views.badge.Badge
+import com.kylecorry.andromeda.views.list.ListItem
+import com.kylecorry.andromeda.views.list.ListItemTag
+import com.kylecorry.andromeda.views.list.ResourceListIcon
+import com.kylecorry.trail_sense.R
+import com.kylecorry.trail_sense.shared.extensions.TrailSenseReactiveBottomSheetFragment
+import com.kylecorry.trail_sense.shared.navigateWithAnimation
+import com.kylecorry.trail_sense.shared.views.SearchView
+import com.kylecorry.trail_sense.tools.survival_guide.infrastructure.SurvivalGuideSearch
+
+class SurvivalGuideBottomSheetFragment :
+    TrailSenseReactiveBottomSheetFragment(R.layout.fragment_survival_guide_bottom_sheet) {
+
+    override fun update() {
+        // Views
+        val searchView = useView<SearchView>(R.id.search)
+        val summaryView = useView<TextView>(R.id.summary)
+        val summaryHolderView = useView<View>(R.id.summary_holder)
+        val summaryTitleView = useView<TextView>(R.id.summary_title)
+        val summaryScrollView = useView<NestedScrollView>(R.id.summary_scroll)
+        val summaryChapterBadgeView = useView<Badge>(R.id.summary_chapter_title)
+        val titleButtonView = useView<AppCompatImageButton>(R.id.survival_guide_btn)
+        val emptyTitleView = useView<TextView>(R.id.empty_view_title)
+        val emptyDescView = useView<TextView>(R.id.empty_view_description)
+        val navController = useNavController()
+
+        // State
+        val (query, setQuery) = useState("")
+        val (searchResults, summary) = useSearchSurvivalGuide(query)
+
+        // Services
+        val context = useAndroidContext()
+        val markdown = useService<MarkdownService>()
+
+        // Navigation
+        titleButtonView.setOnClickListener {
+            dismiss()
+            navController.navigateWithAnimation(
+                R.id.fragmentToolSurvivalGuideList,
+                bundleOf("search_query" to query)
+            )
+        }
+
+        // Colors
+        titleButtonView.imageTintList = ColorStateList.valueOf(Resources.androidTextColorPrimary(requireContext()))
+        val textColor = Resources.androidTextColorSecondary(requireContext())
+        emptyTitleView.setTextColor(textColor)
+        emptyDescView.setTextColor(textColor)
+
+        useSearch(searchView, setQuery)
+
+        useMemo(navController, query, searchResults) {
```

Comment:
It doesn't look like this is needed, can you confirm?

Reply:
Everything else looks good, so i'm going to merge this in and then remove that useMemo if not needed. Thanks!